tests/Rules/Fixture/laravel-view-function.php

```diff
@@ -47,3 +47,32 @@
 ];
 
 view('foo', $fooBar);
+
+class MyDto implements \Illuminate\Contracts\Support\Arrayable {
+    public string $foo;
+
+    public function __construct(
+        string $foo
+    ) {
+        $this->foo = $foo;
+    }
+
+    /**
+     * @return array{foo: string}
+     */
+    public function toArray() {
+        return ['foo' => $this->foo];
+    }
+}
+
+$dataDto = new MyDto('bar');
+
+view('foo', $dataDto);
+
+$dataAnonymousClass = new class extends MyDto {
+    public function __construct() {
+        parent::__construct('bar');
+    }
+};
+
+view('foo', $dataAnonymousClass);
```

Comment:
This case can be expressed more minimally
```suggestion
class MyDto implements \Illuminate\Contracts\Support\Arrayable {
    /**
     * @return array{foo: string}
     */
    public function toArray() {
        return ['foo' => 'bar'];
    }
}

view('foo', new MyDto());
```